packages/fetchai/skills/tac_control_contract/behaviours.py

```diff
@@ -309,12 +320,12 @@ def _create_items(self, token_ids: Union[List[int], int]) -> TransactionMessage:
                 token_id=token_ids,
             )
 
-    def _mint_objects(self, is_batch: bool, address: Address, token_id: int = None):
+    def _mint_objects(self, is_batch: bool, address: Address):
         self.context.logger.info("Minting the items")
         contract = self.context.contracts.erc1155
         parameters = cast(Parameters, self.context.parameters)
         if is_batch:
-            minting = [parameters.base_good_endowment] * parameters.nb_goods
+            minting = [parameters.base_good_endowment] * (parameters.nb_goods - 1)
```

Comment:
The nb_goods that we are passing from the parameter has value `11`. It contains the game currency. The minting list must be of size `10`. We are minting the game currency later on a separate transaction.

Reply:
I think parameter value should still be 10 for goods. It is clear that there must be a single game currency.